clients/venice-samza/src/main/java/com/linkedin/venice/samza/VeniceSystemProducer.java

```diff
@@ -744,6 +770,38 @@ private static <T> byte[] serializePrimitive(T input, DatumWriter<T> writer) {
     return out.toByteArray();
   }
 
+  private Object convertWriteComputeRecordToFullPut(
+      int derivedSchemaId,
+      int baseSchemaId,
+      Object incomingWriteValueObject) {
+    Schema baseSchema = valueSchemaIds.entrySet()
+        .stream()
+        .filter(entry -> entry.getValue().getFirst() == derivedSchemaId && entry.getValue().getSecond() == -1)
+        .map(Map.Entry::getKey)
+        .findFirst()
+        .orElse(null);
+
+    if (baseSchema == null) {
+      // refresh from venice once since we don't have this schema cached yet, then check again
+      this.refreshSchemaCache();
+      baseSchema = valueSchemaIds.entrySet()
+          .stream()
+          .filter(entry -> entry.getValue().getFirst() == derivedSchemaId && entry.getValue().getSecond() == -1)
+          .map(Map.Entry::getKey)
+          .findFirst()
+          .orElse(null);
+      if (baseSchema == null) {
+        // Something isn't right with this write. We can't seem to find an associated schema, so raise an exception.
+        throw new SamzaException(
+            "Unable to find base schema with id: " + baseSchemaId + " for write compute schema with id "
+                + derivedSchemaId);
+      }
+    }
+
+    WriteComputeHandlerV1 writeComputeHandlerV1 = new WriteComputeHandlerV1();
```

Comment:
Since this handler is stateless, we can move it to static field so it could be reused?

Reply:
Good idea, can do.